apps/gp2-frontend/src/AuthenticatedApp.tsx

```diff
@@ -35,22 +40,33 @@ const AuthenticatedApp: FC<Record<string, never>> = () => {
     // order by the likelyhood of user navigating there
     user?.onboarded ? loadOnboardedApp() : loadOnboarding();
   }, [user?.onboarded]);
+  const { addNotification } = useNotificationContext();
 
   const [showWelcomeBackBanner, setShowWelcomeBackBanner] = useState(
     user?.onboarded || false,
   );
+
+  const welcomeBackMessage = `Welcome back to the GP2 Hub${
+    user?.firstName ? `, ${user.firstName}` : ''
+  }!`;
+
+  useEffect(() => {
+    if (showWelcomeBackBanner) {
+      addNotification({
```

Comment:
Is this a hook? If so, it shouldn't be called conditionally?

Reply:
the hook is `useNotificationContext`, the context has notifications plus functions to add/remove them